tests/scripts/thread-cert/base_v1_2_LowPower_6_1_07_PreferringARouterOverAReed.py

```diff
@@ -0,0 +1,156 @@
+            'name': 'DUT',
+            'panid': 0xface,
+            'allowlist': [ROUTER_1, REED_1],
+        },
+    }
+
+    def test(self):
+        print("Run test")
+        self.nodes[LEADER].start()
+        self.simulator.go(5)
+        self.assertEqual(self.nodes[LEADER].get_state(), 'leader')
+
+        self.nodes[ROUTER_1].start()
+        self.simulator.go(5)
+        self.assertEqual(self.nodes[ROUTER_1].get_state(), 'router')
+
+        self.nodes[REED_1].start()
+        self.simulator.go(5)
+        self.assertEqual(self.nodes[REED_1].get_state(), 'child')
+
+        self.simulator.go(5)  # wait for advertisements
+
+        # Step 2 - set outbound link quality of ROUTER_1 and REED_1 to 2 (medium)
+        self.nodes[ROUTER_1].set_outbound_link_quality(2)
+        self.nodes[REED_1].set_outbound_link_quality(2)
+
+        # Step 3 - DUT Automatically begins attach process by sending a multicast MLE Parent Request
+        self.nodes[DUT].start()
+        self.simulator.go(5)
+        self.assertEqual(self.nodes[DUT].get_state(), 'child')
+
+    def verify(self, pv):
+        print("Run verify")
+        pkts = pv.pkts
+        pv.summary.show()
+        LEADER = pv.vars['LEADER']
+        ROUTER_1 = pv.vars['ROUTER_1']
+        REED_1 = pv.vars['REED_1']
+        DUT = pv.vars['DUT']
+
+        # Step 1 - Ensure all routers and leader are sending MLE advertisements
+        pkts.filter_wpan_src64(LEADER) \
+            .filter_mle_cmd(consts.MLE_ADVERTISEMENT) \
+            .must_next()
+        pkts.filter_wpan_src64(ROUTER_1) \
+            .filter_mle_cmd(consts.MLE_ADVERTISEMENT) \
+            .must_next()
+
+        # Step 3 - DUT begins attach process by sending a multicast MLE Parent Request
+        pkts.filter_wpan_src64(DUT) \
+            .filter_mle_cmd(consts.MLE_PARENT_REQUEST) \
+            .filter_mle_has_tlv([TlvType.CHALLENGE, TlvType.MODE]) \
+            .filter(lambda p: p.mle.tlv.scan_mask.r == 1 and p.mle.tlv.scan_mask.e == 0) \
+            .filter(lambda p: p.mle.tlv.version == 3) \
+            .filter(lambda p: p.ipv6.hlim == 255) \
+            .filter_LLARMA() \
+            .must_next()
+
+        # Step 4 - Router_1 automatically Respond with MLE Parent Response
+        pkts.filter_wpan_src64(ROUTER_1) \
+            .filter_wpan_dst64(DUT) \
+            .filter_mle_cmd(consts.MLE_PARENT_RESPONSE) \
+            .filter(lambda p: p.mle.tlv.version == 2) \
+            .must_next()
+
+        # Step 5 - DUT sends another multicast MLE Parent Request to the all-routers multicast with the Scan Mask TLV
+        # set for all routers and all REEDs
+        pkts.filter_wpan_src64(DUT) \
+            .filter_mle_cmd(consts.MLE_PARENT_REQUEST) \
+            .filter_mle_has_tlv([TlvType.CHALLENGE, TlvType.MODE]) \
+            .filter(lambda p: p.mle.tlv.scan_mask.r == 1 and p.mle.tlv.scan_mask.e == 1) \
+            .filter(lambda p: p.mle.tlv.version == consts.THREAD_VERSION_TLV_VALUE_1_2) \
```

Comment:
```suggestion
            .filter(lambda p: p.mle.tlv.version == consts.THREAD_VERSION_1_2) \
```
I think this is the thread version.